Makefile

```diff
@@ -28,10 +28,23 @@ else
   ponydir ?= $(prefix)/lib/pony/$(tag)
 endif
 
+# Use clang by default
+ifndef CC
+  ifneq (,$(shell clang --version 2>&1 | grep 'clang version'))
+    CC = clang
+    CXX = clang++
+  endif
+else ifeq ($(CC), cc)
+  ifneq (,$(shell clang --version 2>&1 | grep 'clang version'))
+    CC = clang
+    CXX = clang++
+  endif
+endif
+
 # By default, CC is cc and CXX is g++
 # So if you use standard alternatives on many Linuxes
 # You can get clang and g++ and then bad things will happen
-ifneq (,$(shell $(CC) --version 2>&1 | grep clang))
+ifneq (,$(shell $(CC) --version 2>&1 | grep 'clang version'))
```

Comment:
what's the clang version change?

i know what i had previously worked for what i tested. unfortunately i don't remember what i tested so i have no idea if this change still covers whatever i tested.

Reply:
On Ubuntu if you call `clang --version` when clang is not installed, it prints some helpful info about running `apt install clang`, which contains the string "clang", so the makefile thinks that clang is installed when it is not.  Searching for "clang version" in the output only matches the actual output of `clang --version`.

Reply:
Makes sense. I definitely didn't test that. Nice catch!

Reply:
I think it would be good to update the comments around the checks to explain why "clang version" rather than just "clang".